core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/controlplane/services/secret/SecretEventListener.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c) 2024 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.edc.connector.controlplane.services.secret;
+
+import org.eclipse.edc.connector.secret.spi.event.SecretCreated;
+import org.eclipse.edc.connector.secret.spi.event.SecretDeleted;
+import org.eclipse.edc.connector.secret.spi.event.SecretEvent;
+import org.eclipse.edc.connector.secret.spi.event.SecretUpdated;
+import org.eclipse.edc.connector.secret.spi.observe.SecretListener;
+import org.eclipse.edc.spi.event.EventEnvelope;
+import org.eclipse.edc.spi.event.EventRouter;
+import org.eclipse.edc.spi.types.domain.secret.Secret;
+
+import java.time.Clock;
+
+/**
+ * Listener responsible for creating and publishing events regarding Secret state changes
+ */
+public class SecretEventListener implements SecretListener {
+    private final Clock clock;
+    private final EventRouter eventRouter;
+
+    public SecretEventListener(Clock clock, EventRouter eventRouter) {
+        this.clock = clock;
+        this.eventRouter = eventRouter;
+    }
+
+    @Override
+    public void created(Secret secret) {
+        var event = SecretCreated.Builder.newInstance()
+                .secretId(secret.getId())
+                .build();
+
+        publish(event);
+    }
+
+    @Override
+    public void deleted(String secretId) {
```

Comment:
```suggestion
    public void deleted(Secret secret) {
```

let's keep it the same as for the other listeners